std/datetime/date.d

```diff
@@ -8966,15 +8966,32 @@ public:
 
 
     /++
-        Converts this $(LREF TimeOfDay) to a string with the format HHMMSS.
+        Converts this $(LREF TimeOfDay) to a string with the format `HHMMSS`.
+        If `writer` is set, the resulting string will be written directly to it.
+
+        Params:
+            writer = A `char` accepting $(REF_ALTTEXT output range, isOutputRange, std, range, primitives)
+        Returns:
+            A `string` when not using an output range; `void` otherwise.
       +/
     string toISOString() const @safe pure nothrow
     {
-        import std.format : format;
+        import std.array : appender;
+        auto w = appender!string();
+        w.reserve(6);
+        toISOString(w);
+        return w.data;
+    }
+
+    /// ditto
+    void toISOString(W)(ref W writer) const
+    if (isOutputRange!(W, char))
+    {
+        import std.format : formattedWrite;
         try
-            return format("%02d%02d%02d", _hour, _minute, _second);
+            formattedWrite(writer, "%02d%02d%02d", _hour, _minute, _second);
         catch (Exception e)
-            assert(0, "format() threw.");
+            assert(0, "formattedWrite() threw.");
```

Comment:
### Off-topic

FYI there's a trick to avoid the red coverage here - from https://github.com/dlang/druntime/pull/2057#discussion_r163652061:

```d
scope (failure) assert(0, "formattedWrite() threw.");
formattedWrite(writer, "%02d%02d%02d", _hour, _minute, _second);
```

Also I wonder whether `formattedWrite` can be made `nothrow` when the input string is passed in at CTFE and thus checked at compile-time.

Reply:
https://issues.dlang.org/show_bug.cgi?id=18304